pkg/resources/scaleway/resources.go

```diff
@@ -0,0 +1,267 @@
+package scaleway
+
+import (
+	"fmt"
+	"k8s.io/kops/pkg/resources"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/scaleway"
+	"strings"
+
+	domain "github.com/scaleway/scaleway-sdk-go/api/domain/v2beta1"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/api/lb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	resourceTypeDNSRecord    = "dns-record"
+	resourceTypeLoadBalancer = "load-balancer"
+	resourceTypeVolume       = "volume"
+	resourceTypeServer       = "server"
+
+	KopsDomainName = "scaleway-terraform.com" //TODO: replace with real domain name later
+)
+
+type listFn func(fi.Cloud, string) ([]*resources.Resource, error)
+
+func ListResources(cloud scaleway.ScwCloud, clusterName string) (map[string]*resources.Resource, error) {
+	resourceTrackers := make(map[string]*resources.Resource)
+
+	listFunctions := []listFn{
+		listDNSRecords,
+		listLoadBalancers,
+		listServers,
+		listVolumes,
+	}
+
+	for _, fn := range listFunctions {
+		rt, err := fn(cloud, clusterName)
+		if err != nil {
+			return nil, err
+		}
+		for _, t := range rt {
+			resourceTrackers[t.Type+":"+t.ID] = t
+		}
+	}
+
+	return resourceTrackers, nil
+}
+
+func listDNSRecords(cloud fi.Cloud, clusterName string) ([]*resources.Resource, error) {
+	c := cloud.(scaleway.ScwCloud)
+	records, err := c.DomainService().ListDNSZoneRecords(&domain.ListDNSZoneRecordsRequest{
+		DNSZone: KopsDomainName,
+		//Type: "A", // DO only looks for records of type A, is it the same for us ?
```

Comment:
Either using an A record on a flexible IP. Or a CNAME with the public DNS name of an instance: 11111111-1111-1111-1111-111111111111.pub.instances.scw.cloud